tests_func/godogs_test.go

```diff
@@ -0,0 +1,391 @@
+	s.Step(`^mongodb auth initialized on mongodb(\d+)$`, authenticateOnMongodb)
+	s.Step(`^mongodb(\d+) has test mongodb data test(\d+)$`, fillMongodbWithTestData)
+	s.Step(`^we create mongodb(\d+) backup$`, createMongodbBackup)
+	s.Step(`^we create mongodb(\d+) backup with user data$`, createMongodbBackupWithUserData)
+	s.Step(`^we got (\d+) backup entries of mongodb(\d+)$`, testBackupEntriesOfMongodb)
+	s.Step(`^we put empty backup via minio(\d+)$`, putEmptyBackupViaMinio)
+	s.Step(`^we delete backups retain (\d+) via mongodb(\d+)$`, deleteBackupsRetainViaMongodb)
+	s.Step(`^we check if empty backups were purged via minio(\d+)$`, testEmptyBackupsViaMinio)
+
+	s.Step(`^we delete #(\d+) backup via mongodb(\d+)$`, deleteBackupViaMongodb)
+	s.Step(`^we restore #(\d+) backup to mongodb(\d+)$`, restoreBackupToMongodb)
+	s.Step(`^we got same mongodb data at mongodb(\d+) mongodb(\d+)$`, testEqualMongodbDataAtMongodbs)
+	s.Step(`^we ensure mongodb(\d+) #(\d+) backup metadata contains$`, mongodbBackupMetadataContainsUserData)
+
+}
+
+func mongodbBackupMetadataContainsUserData(mongodbId int, backupId int, data *gherkin.DocString) error {
+	nodeName := fmt.Sprintf("mongodb%02d.test_net_%s", mongodbId, testUtils.GetVarFromEnvList(testContext.Env, "TEST_ID"))
+	backupList, err := testHelper.GetBackups(testContext, nodeName)
+	if err != nil {
+		return err
+	}
+	if len(backupList)-1-backupId < 0 {
+		return fmt.Errorf("cannot get %dth backup - there are only %d", backupId, len(backupList))
+	}
+	backup := backupList[len(backupList)-1-backupId]
+	path := fmt.Sprintf("/export/dbaas/mongodb-backup/test_uuid/test_mongodb/basebackups_005/%s_backup_stop_sentinel.json", backup)
+	cmd := []string{"cat", path}
+	jsonString, _ := testHelper.RunCommandInContainer(testContext, fmt.Sprintf("minio01.test_net_%s", testUtils.GetVarFromEnvList(testContext.Env, "TEST_ID")), cmd)
+	jsonString = jsonString[strings.Index(jsonString, "{"):]
+
+	var rawUserData map[string]*json.RawMessage
+	err = json.Unmarshal([]byte(jsonString), &rawUserData)
+	if err != nil {
+		return err
+	}
+
+	var userData map[string]*json.RawMessage
+	err = json.Unmarshal(*rawUserData["UserData"], &userData)
+	if err != nil {
+		return err
+	}
+
+	var labels map[string]string
+	err = json.Unmarshal(*userData["labels"], &labels)
+	if err != nil {
+		return err
+	}
+
+	content := getMakeBackupContentFromDocString(data)
+
+	if !reflect.DeepEqual(labels, content["labels"]) {
+		return fmt.Errorf("error: expected labels don't equal to existing in backup")
+	}
+
+	return nil
+}
+
+func testMongodbConnect(mongodbId int) error {
+	nodeName := fmt.Sprintf("mongodb%02d.test_net_%s", mongodbId, testUtils.GetVarFromEnvList(testContext.Env, "TEST_ID"))
+	for i := 0; i < 25; i++ {
+		connection, _ := testHelper.EnvDBConnect(testContext, nodeName)
+		err := connection.Database(nodeName).Client().Ping(testContext.Context, nil)
+		if err == nil {
+			return nil
+		}
+		time.Sleep(100 * time.Millisecond)
```

Comment:
const timeoutInterval := 100 * time.Millisecond will be better

Reply:
fixed